app/controllers/pair_feedbacks_controller.rb

```diff
@@ -22,19 +19,16 @@ def new
 
   def create
     @student = current_student
-    today = Time.zone.now.to_date
-    if @student.pairs_on_day(today).any?
+    if @student.pairs_without_feedback_today.any?
       @pair_feedback = PairFeedback.new(pair_feedback_params)
-      pairs = @student.pairs_on_day(today)
       if @pair_feedback.save
-        @pairs_without_feedback = pairs.select { |pair| !PairFeedback.where(created_at: today.all_day).find_by(student: @student, pair: pair)}
-        if @pairs_without_feedback.any?
+        if @student.pairs_without_feedback_today.any?
           redirect_to sign_out_path
         else
           sign_out_student
         end
       else
-        @pairs_without_feedback = pairs.select { |pair| !PairFeedback.where(created_at: today.all_day).find_by(student: @student, pair: pair)}
+        @pairs_without_feedback = @student.pairs_without_feedback_today
         render :new
```

Comment:
Maybe redirect to /new instead of rendering :new. That way the user can hit refresh or bookmark the page and the page will load properly.

Reply:
I think the render :new is there so that it will render validation error messages, but really the only reason it'll fail is if one or more of the questions is unanswered, so I could just redirect with an alert. It will simplify the code too to use a redirect. I think I was just under the impression it was better to render the validation error messages when possible.

Reply:
Oh, no, you're right, I just misunderstood the code. It is generally better to render :new with the validation error messages. That way the form is already filled out (if there is one) from the instance variables' data.